sssd_test_framework/roles/ipa.py

```diff
@@ -396,6 +408,85 @@ def modify(
         self._modify(attrs, input=password)
         return self
 
+    def passkey_add(self, passkey_mapping: str) -> IPAUser:
+        """
+        Add passkey mapping to the user.
+
+        :param passkey_mapping: Passkey mapping generated by ``sssctl passkey-register``
+        :type passkey_mapping: str
+        :return: Self.
+        :rtype: IPAUser
+        """
+        self._exec("add-passkey", [passkey_mapping])
+        return self
+
+    def passkey_add_register(
+        self,
+        *,
+        pin: str | int,
+        device: str,
+        ioctl: str,
+        script: str,
+    ) -> str:
+        """
+        Register passkey with the user (run ipa user-add-passkey --register).
+
+        :param pin: Passkey PIN.
+        :type pin: str | int
+        :param device: Path to local umockdev device file.
+        :type device: str
+        :param ioctl: Path to local umockdev ioctl file.
+        :type ioctl: str
+        :param script: Path to local umockdev script file
+        :type script: str
+        :return: Generated passkey mapping string.
+        :rtype: str
+        """
+        device_path = self.role.fs.upload_to_tmp(device, mode="a=r")
+        ioctl_path = self.role.fs.upload_to_tmp(ioctl, mode="a=r")
+        script_path = self.role.fs.upload_to_tmp(script, mode="a=r")
+
+        command = self.role.fs.mktmp(
+            rf"""
+            #!/bin/bash
+
+            LD_PRELOAD=/opt/random.so umockdev-run \
+                --device '{device_path}'                \
+                --ioctl '/dev/hidraw1={ioctl_path}'     \
+                --script '/dev/hidraw1={script_path}'   \
+                -- ipa user-add-passkey '{self.name}' --register --cose-type=es256 --require-user-verification=True
+        """,
+            mode="a=rx",
+        )
+
+        result = self.host.ssh.expect(
+            f"""
+            set timeout 40
```

Comment:
Could we use a default argument for the timeout here?

Reply:
Done.